src/main/java/uk/gov/indexer/Application.java

```diff
@@ -23,15 +24,33 @@ public static void main(String[] args) throws IOException, SQLException, Interru
         Configuration configuration = new Configuration(args);
         Set<String> registers = configuration.getRegisters();
 
-        ScheduledExecutorService executorService = Executors.newScheduledThreadPool(registers.size());
+        ScheduledExecutorService executorService = Executors.newScheduledThreadPool((int) registers.stream().filter(r -> configuration.cloudSearchEndPoint(r).isPresent()).count() + registers.size());
+
         addShutdownHook(executorService);
 
         for (String register : registers) {
             try {
-                DestinationDBUpdateDAO destinationQueryDAO = createDestinationDAO(register, configuration);
-                SourceDBQueryDAO sourceQueryDAO = createSourceDAO(register, configuration);
+                DBI dbi = new DBI(configuration.getProperty(register + ".destination.postgres.db.connectionString"));
+
+                DestinationDBUpdateDAO destinationDBUpdateDAO = createDestinationDBUpdateDAO(dbi);
+
+                executorService.scheduleAtFixedRate(
+                        new IndexerTask(register, createSourceDAO(register, configuration), destinationDBUpdateDAO),
+                        0,
+                        10,
+                        TimeUnit.SECONDS
+                );
+
+                configuration.cloudSearchEndPoint(register).ifPresent(
+                        endPoint -> executorService.scheduleAtFixedRate(
+                                new CloudSearchDataUploadTask(register, endPoint, configuration.cloudSearchWaterMarkEndPoint(register).get(), dbi.onDemand(IndexedEntriesUpdateDAO.class)),
+                                0,
+                                10,
```

Comment:
Discussed! 10 seconds are fine as it is a wait time in successive execution of task.